### ISSUE TITLE
Ruff fails to detect invalid Python syntax in f-string with single quotes

### ISSUE THREAD

**Issue by mgaitan** [astral-sh/ruff#13322]
I found an issue where Ruff doesn't detect a syntax error in f-string. Consider this simple source:

```python
import os

location = f's3://{os.environ.get('bucket', 'dev')}'
```

Ruff reports no issues:

```bash
$ uv tool run ruff check ruff_bug.py
All checks passed!
$ uv tool run ruff format ruff_bug.py
1 file left unchanged
```

However, the module is invalid: 

```bash
$ python ruff_bug.py
  File "ruff_bug.py", line 3
    location = f's3://{os.environ.get('bucket', 'dev')}'
                                       ^
SyntaxError: invalid syntax
```

Tested in ruff 0.5.1 and 0.6.4.  It seems like Ruff isn't catching this invalid syntax with mismatched quotes in the f-string. 

Is this an expected behaviour?

**Comment by MichaReiser** [comment]
Yes, and no. 

It's expected behavior because this is a known limitation of Ruff and there have been a few related issues like https://github.com/astral-sh/ruff/issues/10163 

But no, because we agree that this is not a good user experience. The problem is that our parser currently doesn't respect the project's target version. It always uses the latest syntax which allows using the same quotes for the f-string and inner expressions. There's an issue to either make the parser target-version aware OR create a lint rule that flags all syntax that isn't compatible with the project's target version. 

https://github.com/astral-sh/ruff/issues/6591
